source/Engine.cpp

```diff
@@ -319,8 +319,7 @@ void Engine::Place()
 			// Default to the player's planet in the case of data definition errors.
 			if(person.IsLaunching() || launchesWithPlayer || (hasOwnPlanet && !object))
 			{
-				if(planet)
-					ship->SetPlanet(planet);
+				ship->SetPlanet(planet);
```

Comment:
[Commit](https://github.com/endless-sky/endless-sky/pull/7917/commit/c7db982157ba91df73cb6b16e69a664c89051de2): `planet` come from `player.GetPlanet()` and cppcheck suppose isn't a nullptr deference until it gets to this line. At that point it raises an issue because it can't be sure anymore: why you check, isn't? ...So (grossly) cppcheck reasoning is that any pointer is not null until checked.

But the same variable (`planet`) is used even if null: see [this line](https://github.com/endless-sky/endless-sky/blob/1d666509d5edf428cc2bce5e3046574e22078e43/source/Engine.cpp#L315) and I conclude it couldn't be... I may be wrong though...

Reply:
We should probably add the null check above when planet is used instead of removing the null checks altogether. (so like `(planet && planet.CanLand(...))`. I'm pretty sure it should never be null, but still.

Reply:
Done, though I don't feel it's completely closed :P There's also [this](https://github.com/endless-sky/endless-sky/blob/8ed603dec83ef8337669e5787811b3a305ec7571/source/Engine.cpp#L342), which is relevant. Indeed having a `if (planet) ship.SetPlanet(planet)` seems more a _wanted functionality_ which is registered as a cppcheck warning for different reasons than an error: if `planet` is null, there's no problem here.

However, I'm just taking some note here, as the above is only a though I had...